### DISCUSSION TITLE
Cant seem to add a second nvidia shield to music assistant?

### DISCUSSION THREAD

**Discussion by danktankk** [music-assistant/support#2148] [Q&A] [answered]
As the title says, I have 2 shields and they are both already integrated into home assistant and work great.  I have them both listed in the android remote integration plus th google cast integration.  For whatever reason, music assistant only adds one of the two, I cant seem to get the second one added.

Any idea how to fix this?

**Comment by OzGav** [comment]
Look in the logs when MA starts for discovery info. Are they both on the same subnet?

**Reply by danktankk** [reply]
The nvidia shields are on 192.168.20.0 (192.168.20.4 & 192.168.20.5) and Home assistant is on 192.168.160.0 network with a static IP.    I didnt mention that it also found my xgimi projector, I assume because it uses android, but I never use that and change toi HDMI 2 where the shield is plugged in.

**Reply by OzGav** [reply, marked as answer]
As per the docs the devices must be on the same subnet as MA for discovery to work

**Comment by danktankk** [comment]
It appears that is not the case for 2 of 3 devices.  No worries though.  I can just use Spotify or whatever.  Thanks

**Reply by danktankk** [reply]
It would be nice to be able to manually add devices, but it's not that important for me personally.

**Reply by OzGav** [reply]
Why it works at all is a mystery unless your routers are doing some sort of mDNS forwarding. We make it very clear that this is not supported. It is not necessary to manually add devices if they are on the same network as they will be discovered.

**Reply by marcelveldt** [reply]
mdns forwarders are pure evil and should be avoided at all times. Best case scenario they actually manage to forward some of the mdns packets but worst case (and we see that a lot) they malform packets, reverse them, cause loops or just flood the network with traffic. Just dont use them and make sure your IoT gear is on the same vlan/subnet as HA/MA.
